src/libraries/System.Private.CoreLib/src/System/Half.cs

```diff
@@ -0,0 +1,700 @@
+        /// <summary>Determines whether the specified value is finite (zero, subnormal, or normal).</summary>
+        public static bool IsFinite(Half value)
+        {
+            return StripSign(value) < PositiveInfinityBits;
+        }
+
+        /// <summary>Determines whether the specified value is infinite.</summary>
+        public static bool IsInfinity(Half value)
+        {
+            return StripSign(value) == PositiveInfinityBits;
+        }
+
+        /// <summary>Determines whether the specified value is NaN.</summary>
+        public static bool IsNaN(Half value)
+        {
+            return StripSign(value) > PositiveInfinityBits;
+        }
+
+        /// <summary>Determines whether the specified value is negative.</summary>
+        public static bool IsNegative(Half value)
+        {
+            return (short)(value._value) < 0;
+        }
+
+        /// <summary>Determines whether the specified value is negative infinity.</summary>
+        public static bool IsNegativeInfinity(Half value)
+        {
+            return value._value == NegativeInfinityBits;
+        }
+
+        /// <summary>Determines whether the specified value is normal.</summary>
+        // This is probably not worth inlining, it has branches and should be rarely called
+        public static bool IsNormal(Half value)
+        {
+            uint absValue = StripSign(value);
+            return (absValue < PositiveInfinityBits)    // is finite
+                && (absValue != 0)                      // is not zero
+                && ((absValue & ExponentMask) != 0);    // is not subnormal (has a non-zero exponent)
+        }
+
+        /// <summary>Determines whether the specified value is positive infinity.</summary>
+        public static bool IsPositiveInfinity(Half value)
+        {
+            return value._value == PositiveInfinityBits;
+        }
+
+        /// <summary>Determines whether the specified value is subnormal.</summary>
+        // This is probably not worth inlining, it has branches and should be rarely called
+        public static bool IsSubnormal(Half value)
+        {
+            uint absValue = StripSign(value);
+            return (absValue < PositiveInfinityBits)    // is finite
+                && (absValue != 0)                      // is not zero
+                && ((absValue & ExponentMask) == 0);    // is subnormal (has a zero exponent)
+        }
+
+        /// <summary>
+        /// Parses a <see cref="Half"/> from a <see cref="string"/> in the default parse style.
+        /// </summary>
+        /// <param name="s">The input to be parsed.</param>
+        /// <returns>The equivalent <see cref="Half"/> value representing the input string. If the input exceeds Half's range, a <see cref="Half.PositiveInfinity"/> or <see cref="Half.NegativeInfinity"/> is returned. </returns>
+        public static Half Parse(string s)
+        {
+            return (Half)float.Parse(s, style: DefaultParseStyle, provider: null);
```

Comment:
These should be updated to call into `Number.ParseHalf` methods (which pass in the right constants) rather than going through `float`